### ISSUE TITLE
[css-flexbox][align-items] Support for aligning items to the center of the first line.

### ISSUE THREAD

**Issue by jmiguelrivas** [w3c/csswg-drafts#9854]
## Spec links:
[align-items-property](https://www.w3.org/TR/css-flexbox-1/#align-items-property)

## Premise:
I'm recently finding myself in this situation where I want to align a checkbox or a help icon with the first line of the content.

but it seems that the only way to achieve something like this would implied making a container around the input and align visually the checkbox with the text (which could later break when the `font-size`/`line-height` change)

## Proposal:

would it make sense having a property under `flexbox`, something like `align-items: first-line-center` for these cases?

[codepen](https://codepen.io/koturno/pen/dyrVJGx?editors=1100)

![image](https://github.com/w3c/csswg-drafts/assets/38018741/b44a49ae-3f6f-455b-8db3-07d0fcbcb854)
![image](https://github.com/w3c/csswg-drafts/assets/38018741/5e7a660e-6a42-4f03-a512-bfd57dfede16)

**Comment by fantasai** [comment]
I'm not sure I understand. Are you looking for `align-self: first baseline`?

**Comment by jmiguelrivas** [comment]
> I'm not sure I understand. Are you looking for `align-self: first baseline`?

Thanks for your response :)

Not quite baseline but center relative to the first line, I'm able to use align center in some cases but it fails when I got a multiline paragraph (which tends to happens when the screen get smaller and there is less space on the container)

Single line:
![image](https://github.com/w3c/csswg-drafts/assets/38018741/b3f1000d-6274-41c0-8e1d-d8a71e9c08c1)

Multiline:
![image](https://github.com/w3c/csswg-drafts/assets/38018741/57662057-f0ff-424b-9451-5ea546fc6356)


This is the scenario that I'm trying to cover:

![image](https://github.com/w3c/csswg-drafts/assets/38018741/f5a175df-0075-45c5-ac5e-d6b2fb80b603)

**Comment by adamkudrna** [comment]
Checkboxes and leading icons are a classical use case. I'm solving the same issue right now in our [design system](https://github.com/lmc-eu/spirit-design-system/blob/feature/web-toast/packages/web/src/scss/components/Toast/_ToastBar.scss):

I need the icon to **visually** stick to the center of the first line which is only possible when the height of the icon is the same as the line height of the text (which is luckily our case, so no padding tweaks are needed). Then both elements can be **technically** aligned to `start`. Exactly for this, an option to align the icon to the center of the first line would prove useful.

<img width="489" alt="obrazek" src="https://github.com/w3c/csswg-drafts/assets/5614085/7f6395fa-e3b9-4986-be68-21ae02377cfe">

**Comment by benface** [comment]
YES. This is such a common problem I run into. Huge +1 from me.

**Comment by knowler** [comment]
Yes, I’d love something like: `align-items: first center` to align elements like icons to the first line of text.

My current solution is to use `align-items: first baseline` to get it on the baseline, then adjust the positioning to centre it with `translate: 0 calc(50% - 0.5cap)` (half the difference between the icon size and text cap size; demo: https://knowler.dev/demos/d_pDxD0?codepen).
